graylog2-web-interface/src/components/common/Spinner.jsx

```diff
@@ -1,22 +1,62 @@
-import React from 'react';
+// @flow strict
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
+import styled, { type StyledComponent } from 'styled-components';
 
 import Icon from './Icon';
 
+const Wrapper: StyledComponent<{ visible: boolean }, {}, HTMLSpanElement> = styled.span`
+  visibility: ${({ visible }) => (visible ? 'visible' : 'hidden')};
+`;
+
+type Props = {
+  delay?: number,
+  fixedWidth?: boolean,
+  name?: string,
+  pulse: boolean,
+  text?: string,
+}
+
 /**
  * Simple spinner to use while waiting for something to load.
  */
-const Spinner = ({ name, text, ...props }) => <span><Icon name={name} spin {...props} /> {text}</span>;
```

Comment:
I had a problem with the usage of {...props} or in my case {...rest} and replaced it with some specific props and prop definitions. i hope I found every use case, so far I only found `pulse` and `fixedWidth`

![image](https://user-images.githubusercontent.com/46300478/75540725-e1acbe00-5a1c-11ea-9c2b-5542486678ce.png)

Reply:
I saw the usage of `{...rest}` in other components (e.g. `views/components/Select.jsx`) where we are using flow as well. I don't really understand why we do not see the same error there.

Reply:
I was testing locally, and I can see the error when the spread happens at the end of the list of props.

This throws the error:
```
<Icon name={name} spin {...props} /> {text}
```

This seems to be fine:
```
<Icon {...props} name={name} spin /> {text}
```

I guess that makes sense, since the rest props could contain a `spin` property that overrides the value, when the spread happens after the prop is set.

Reply:
Thanks for the clarification, this totally makes sense! I will reimplement the `{...props}` because I like the flexibility.

Reply:
I just realized the update of `eslint-sonfig-airbnb` https://github.com/Graylog2/graylog2-server/pull/7586 forbids the props spreading. We need to discuss If we really want this, but the explanation contains some good points. I will not reimplement the props spreading for now.

Reply:
Sounds good, I think for this component is also fine having them explicitly set.